src/libAtomVM/utils.h

```diff
@@ -223,4 +223,14 @@ static inline __attribute__((always_inline)) void *cast_func_to_void_ptr(func_pt
 
 #endif
 
+/*
+ * CONTAINER_OF is useful for obtaining the outer struct.
+ * It makes use of offsetof() from stddef.h.
```

Comment:
Is it equivalent to `GET_LIST_ENTRY` in linkedlist.h? Then it should probably be factored.

Reply:
not equivalent, it is better: this doesn't rely on ((foo *) 0)->bar trick, which is a trick that relies on something out of the standard. Instead it uses offsetof, so `GET_LIST_ENTRY` should be changed to rely on this, however I don't want to go out-of-scope, so, instead I'm going to open a TODO and handle this as a different PR.